### ISSUE TITLE
Filter based on eloquent relationship

### ISSUE THREAD

**Issue by Wotuu** [yajra/laravel-datatables#1818]
### Summary of problem or feature request

I'm trying to filter a list of items based on a relationship. To put in laymans terms, I have a list of dungeonroutes which may have many affixes groups (substitute dungeonroutes with posts, and affix groups with tags if you're confused). I want to show only all dungeonroutes when they have a specific affix group. I have tried many many things with the `addColumn` function, `filterColumn`, `rawColumn` and it just will not budge. Furthermore, rather than returning an object, the relationship is always represented as an encoded json string rather than a normal object *.

I send the back end an array of IDs, like [1, 2, 3], and I want to return all dungeonroutes which has 1 OR 2  OR 3 in their list of affix group IDs.

### Code snippet of problem

Ok this is a bit messy, but it shows what I've tried and probably where I fail:

Relation:
```
    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
     */
    public function affixes()
    {
        return $this->belongsToMany('App\Models\AffixGroup', 'dungeon_route_affix_groups');
    }
```

```
    function list(Request $request)
    {
// I tried filtering the model but that leads nowhere, I get near-empty replies when I try this
        $model = DungeonRoute::query()->with(['affixes'])
//            ->whereHas('affixes', function($query){
//
//        })
        ;
// When I use addColumn the 'affixes' response is now a string containing json, rather than just json.
        return DataTables::eloquent($model)
// I also tried the filterColumn function but I feel like that filters the affixes, as in, it still shows the DungeonRoute but will just strip the DungeonRoute of it's affixes that don't match, I want to remove the entire DungeonRoute if it doesn't match
//            ->addColumn('affixes', function (DungeonRoute $dungeonRoute) {
//                    return $dungeonRoute->affixes->filter(function($affixGroup) {
//                        return false;
////                        if( null !== request('affixes.id') ){
////                            return in_array($affixGroup->id, request('affixes.id'));
////                        } else {
////                            return true;
////                        }
//                    })->all();
//                })
// This helps with not encoding the json string, but it's still json inside a string, I want just json *
//            ->rawColumns(['affixes'])
//            ->make(true)
            ->toJson();

        // @todo this must be the wrong way of doing it
//        $result = datatables(DungeonRoute::with(['dungeon', 'affixes', 'author']))->toArray();
//        unset($result['input']);
//        unset($result['queries']);
//        return json_encode($result);
    }
```

### System details

- Ubuntu 16.04
- 7.1.19
- 5.6
- 1.10.18

I feel like this should be easy but I'm missing something. http://yajrabox.com/docs/laravel-datatables/master/relationships unfortunately does not help me.

* This is what I mean with the nested JSON:
![image](https://user-images.githubusercontent.com/811809/43955234-817f6014-9c9f-11e8-843b-c51d39bc4f5d.png)

**Comment by yajra** [comment]
I think this demo https://datatables.yajrabox.com/eloquent/relationships is what you need. Thanks!

**Comment by Wotuu** [comment]
I've reviewed my solution to hopefully use the Yajra library but so far your link has not proven useful. Most notably https://datatables.yajrabox.com/relation/has-many shows source for `hasOne` functions. I also find the code samples a bit meaningless without some explanatory text.

I did some prodding and poking with your library again but I am still not able to filter anything based on a 1:many relation. If I have a list of Posts, where each post has 0..n Tags, how would I go about it to display a list of posts where I search for 1..n tags? Say I want all posts that are tagged both 'programming' and 'laravel', I don't see how that's currently possible without doing manual processing prior to feeding the data to the library. Perhaps that's the way it should be done, but I had hoped it'd handle it for me in some shape or form.
